src/transmuxer-worker.js

```diff
@@ -207,6 +208,41 @@ class MessageHandlers {
     }, [segment.buffer]);
   }
 
+  /**
+   * Initializes the WebVttParser and passes the init segment.
+   *
+   * @param {Uint8Array} data mp4 boxed WebVTT init segment data
+   */
+  initMp4WebVttParser(data) {
+    if (!this.webVttParser) {
+      this.webVttParser = new WebVttParser();
+    }
+    const segment = new Uint8Array(data.data, data.byteOffset, data.byteLength);
+
+    // set the timescale for the parser.
+    this.webVttParser.init(segment);
+  }
+
+  /**
+   * Parse an mp4 encapsulated WebVTT segment and return an array of cues.
+   *
+   * @param {Uint8Array} data a text/webvtt segment
+   * @return {Object[]} an array of parsed cue objects
+   */
+  getMp4WebVttText(data) {
+    if (!this.webVttParser) {
```

Comment:
Is this check necessary? Mainly asking because I am unclear if the order of the `postMessage` calls will guarantee the parser exists

Reply:
Checking here is only precautionary. Ideally any series of fmp4 text segments should be preceded by an init segment which would initialize the parser and set the timescale. I'm only inclined to keep it here to improve resiliency, maybe logging here would be helpful?

Reply:
Yeah, a warning that explaining that the VTT parser is being init in that function could be useful

Reply:
Actually, looking at this again I'm not sure we want a log here, since this is in the transmuxer worker we wouldn't be able to easily send this through the logger. I'll add a comment instead as clarification. The parser will fallback to the default 90k timescale if not initialized by an init segment anyway, so nothing would break, but the cue timing could potentially be off.